src/main/webapp/app/grading-system/bonus/bonus.service.ts

```diff
@@ -0,0 +1,250 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { GradeStep, GradeStepsDTO } from 'app/entities/grade-step.model';
+import { Bonus, BonusExample, BonusStrategy } from 'app/entities/bonus.model';
+import { GradingScale } from 'app/entities/grading-scale.model';
+import { GradingSystemService } from 'app/grading-system/grading-system.service';
+import { roundValueSpecifiedByCourseSettings } from 'app/shared/util/utils';
+
+export type EntityResponseType = HttpResponse<Bonus>;
+
+@Injectable({ providedIn: 'root' })
+export class BonusService {
+    public resourceUrl = SERVER_API_URL + 'api';
+
+    constructor(private http: HttpClient, private gradingSystemService: GradingSystemService) {}
+
+    /**
+     * Deletes the bonus
+     *
+     * @param bonusId the id of the bonus which will be deleted
+     */
+    deleteBonus(bonusId: number): Observable<HttpResponse<any>> {
+        return this.http.delete<any>(`${this.resourceUrl}/bonus/${bonusId}`, { observe: 'response' });
```

Comment:
💡Idea: Avoid any. You can use `void` here for example

Reply:
Incorporated the suggestion, thanks!